plugins/filter_record_modifier/filter_modifier.c

```diff
@@ -242,6 +250,59 @@ static int make_bool_map(struct record_modifier_ctx *ctx, msgpack_object *map,
     return ret;
 }
 
+static char suffix_key_type(msgpack_packer *tmp_pck, msgpack_object *key, msgpack_object *val)
+{
+    char suffix = 0;
+
+    switch(val->type) {
+    case MSGPACK_OBJECT_NIL:
+    case MSGPACK_OBJECT_ARRAY:
+    case MSGPACK_OBJECT_MAP:
+        break;
+
+    case MSGPACK_OBJECT_BOOLEAN:
+        suffix = 'b';
+        break;
+
+    case MSGPACK_OBJECT_POSITIVE_INTEGER:
+    case MSGPACK_OBJECT_NEGATIVE_INTEGER:
+        suffix = 'i';
+        break;
+    case MSGPACK_OBJECT_FLOAT32:
+    case MSGPACK_OBJECT_FLOAT64:
+        suffix = 'f';
+        break;
+
+    case MSGPACK_OBJECT_STR:
+        suffix = 's';
+        break;
+
+    case MSGPACK_OBJECT_BIN:
+        suffix = 'n';
+        break;
+
+    case MSGPACK_OBJECT_EXT:
+        suffix = 'e';
+        break;
+
+    default:
+        flb_warn("[%s] unknown msgpack type %i", __FUNCTION__, val->type);
+    }
+
+    if ( suffix != 0 ) {
+        int sfsize = key->via.str.size+2;
+        char *skey = flb_strndup(key->via.str.ptr, sfsize);
+        memset(&skey[sfsize-2], '_', 1);
```

Comment:
please avoid memset in this case if you are setting only one byte, do a direct assignation of the value instead.